elyra/pipeline/pipeline_definition.py

```diff
@@ -0,0 +1,281 @@
+        for node in self._node['nodes']:
+            nodes.append(Node(self._pipeline_definition, node))
+
+        return nodes
+
+    def get_property(self, key: str, default_value=None) -> Any:
+        self._validate()
+        return self._node['app_data']['properties'].get(key, default_value)
+
+    def set_property(self, key: str, value: Any):
+        if not key:
+            raise ValueError("Key is required")
+
+        if not value:
+            raise ValueError("Value is required")
+
+        self._node['app_data']['properties'][key] = value
+
+    def validate(self) -> list:
+        validation_issues = list()
+        if 'properties' not in self._node['app_data']:
+            validation_issues.append("Node is missing 'properties' field.")
+        elif len(self._node['app_data']['properties']) == 0:
+            validation_issues.append("Pipeline has zero length 'properties' field.")
+
+        return validation_issues
+
+
+class Node(NodeBase):
+    def __init__(self, pipeline_definition: Dict, node: Dict):
+        super().__init__(pipeline_definition, node)
+
+    @property
+    def type(self) -> str:
+        return self._node.get('type')
+
+    @property
+    def op(self) -> str:
+        return self._node.get('op')
+
+    @property
+    def label(self) -> str:
+        return self._node['app_data']['ui_data'].get('label', self._node['app_data']['label'])
+
+    def get_component_parameter(self, key: str, default_value=None) -> Any:
+        self._validate()
+        return self._node['app_data']['component_parameters'].get(key, default_value)
+
+    def set_component_parameter(self, key: str, value: Any):
+        if not key:
+            raise ValueError("Key is required")
+
+        if not value:
+            raise ValueError("Value is required")
+
+        self._node['app_data']['component_parameters'][key] = value
+
+    def validate(self) -> list:
+        validation_issues = list()
+        if "component_parameters" not in self._node['app_data']:
+            validation_issues.append("Node is missing 'component_parameters' field")
+
+        return validation_issues
+
+
+class PipelineDefinition(object):
+    """
+    Represents a helper class to manipulate pipeline json structure
+    """
+    _primary_pipeline = None
+
+    def __init__(self, pipeline_path: Optional[str] = None, pipeline_definition: Optional[Dict] = None):
+        """
+        The constructor enables either passing a pipeline path or the content of the pipeline definition.
+
+        """
+        if not pipeline_path and not pipeline_definition:
+            # at least one parameter should be provided
+            raise ValueError("At least one parameter must be provided ('pipeline_path' or 'pipeline_definition')")
+        if pipeline_path and pipeline_definition:
+            # only one parameter should be provided
+            raise ValueError("Only one parameter should be provided ('pipeline_path' or 'pipeline_definition')")
```

Comment:
I think one way to handle this exclusive-or kind of behavior would be something like:
```python
    def __init__(self, pipeline_source: Union[str, Dict]):
        if isinstance(pipeline_source, str):
             # supporting loading pipeline from file
            if not os.path.exists(pipeline_source):
                raise ValueError(f"Pipeline file not found: '{pipeline_source}'\n")

            with open(pipeline_source) as f:
                try:
                    self._pipeline_definition = json.load(f)
                except ValueError as ve:
                    raise ValueError(f"Pipeline file is invalid: \n {ve}")
        elif isinstance(pipeline_source, dict):
            # supporting passing the pipeline definition directly
            self._pipeline_definition = pipeline_source
        else:
            raise TypeError(f"Parameter '{pipeline_source}' was not a string or dictionary instance.")
```